packages/gatsby/src/internal-plugins/query-runner/file-parser.js

```diff
@@ -117,8 +117,17 @@ const cache = {}
 
 export default class FileParser {
   async parseFile(file: string): Promise<?DocumentNode> {
-    // TODO figure out why fs-extra isn't returning a promise
-    const text = fs.readFileSync(file, `utf8`)
+    console.warn(`parsing file ${file}`);
+    let text;
+    try {
+      text = await fs.readFile(file, `utf8`)
```

Comment:
Although this change is part of the throwaway commit, I couldn't reproduce the issue mentioned in the TODO. Seems like the conversion to async io "just works."